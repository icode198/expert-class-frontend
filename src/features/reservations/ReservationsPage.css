.reservation-item {
  padding: 1rem;
  border: 1px solid #000;
}
