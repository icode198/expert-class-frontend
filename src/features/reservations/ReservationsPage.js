import './ReservationsPage.css';
import React from 'react';
import { useSelector } from 'react-redux';

export default function ReservationsPage() {
  const { reservations } = useSelector((state) => state.reservations);

  const reservationsList = reservations.map((r) => (
    <div key={r.id} className="reservation-item">
      <h3>{r.class_id}</h3>
      <h4>{r.user_id}</h4>
      <p>{r.date}</p>
      <p>{r.city}</p>
    </div>
  ));

  return (
    <div>
      <h1>Reservations Page</h1>
      {reservationsList}
    </div>
  );
}
