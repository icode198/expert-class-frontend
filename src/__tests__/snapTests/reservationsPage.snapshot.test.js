import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderWithRedux } from '../../tests/test-utils';
import ReservationsPage from '../../pages/reservationsPage/reservationsPage';

afterEach(cleanup);

describe('ReservationsPage', () => {
  test('Renders the page', () => {
    const app = renderWithRedux(<ReservationsPage />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
