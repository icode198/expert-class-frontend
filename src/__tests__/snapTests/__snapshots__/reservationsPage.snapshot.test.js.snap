// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationsPage Renders the page 1`] = `
<div
  className="private-page-container pageContainer"
>
  <h2
    className="heading"
  >
    My Reservations
  </h2>
  <div
    className="reservationsGridContainer"
  />
</div>
`;
